updater/internal/service/service_darwin.go

```diff
@@ -107,7 +125,46 @@ func (d darwinService) Uninstall() error {
 	return nil
 }
 
-// InstallDir returns the filepath to the install directory
-func installDir() (string, error) {
-	return darwinInstallDir, nil
+func (d darwinService) Update() error {
+	if err := d.uninstall(); err != nil {
+		return fmt.Errorf("failed to uninstall old service: %w", err)
+	}
+
+	if err := d.install(); err != nil {
+		return fmt.Errorf("failed to install new service: %w", err)
+	}
+
+	return nil
+}
+
+func (d darwinService) Backup(outDir string) error {
+	oldFile, err := os.Open(d.installedServiceFilePath)
+	if err != nil {
+		return fmt.Errorf("failed to open old service file: %w", err)
+	}
+	defer func() {
+		err := oldFile.Close()
+		if err != nil {
+			log.Default().Printf("darwinService.Backup: failed to close out file: %s", err)
+		}
+	}()
+
+	// Create the file in the specified location; If the file already exists, an error will be returned,
+	// since we don't want to overwrite the file
+	backupFile, err := os.OpenFile(path.BackupServiceFile(outDir), os.O_CREATE|os.O_WRONLY|os.O_EXCL, 0600)
+	if err != nil {
+		return fmt.Errorf("failed to open output file: %w", err)
+	}
+	defer func() {
+		err := backupFile.Close()
+		if err != nil {
+			log.Default().Printf("darwinService.Backup: failed to close out file: %s", err)
+		}
+	}()
+
+	if _, err := io.Copy(backupFile, oldFile); err != nil {
+		return fmt.Errorf("failed to copy service file to backup: %w", err)
+	}
```

Comment:
This code looks almost the same as the code in file.CopyFile. Any reason we don't reuse it?